airflow/providers/amazon/aws/operators/emr_containers.py

```diff
@@ -0,0 +1,151 @@
+    :type sleep_time: int
+    :param max_tries: Maximum number of times to wait for the job run to finish.
+        Defaults to None, which will poll until the job is *not* in a pending, submitted, or running state.
+    :type max_tries: int
+    """
+
+    template_fields = ["name", "virtual_cluster_id", "execution_role_arn", "release_label", "job_driver"]
+    ui_color = "#f9c915"
+
+    @apply_defaults
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        name: str,
+        virtual_cluster_id: str,
+        execution_role_arn: str,
+        release_label: str,
+        job_driver: dict,
+        configuration_overrides: Optional[dict] = None,
+        client_request_token: Optional[str] = None,
+        aws_conn_id: str = "aws_default",
+        sleep_time: int = 30,
+        max_tries: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.name = name
+        self.virtual_cluster_id = virtual_cluster_id
+        self.execution_role_arn = execution_role_arn
+        self.release_label = release_label
+        self.job_driver = job_driver
+        self.configuration_overrides = configuration_overrides or {}
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = client_request_token or str(uuid4())
+        self.sleep_time = sleep_time
+        self.max_tries = max_tries
+        self.job_id = None
+
+    @cached_property
+    def hook(self) -> EMRContainerHook:
+        """Create and return an EMRContainerHook."""
+        return EMRContainerHook(
+            self.aws_conn_id,
+            virtual_cluster_id=self.virtual_cluster_id,
+        )
+
+    def execute(self, context: dict) -> Optional[str]:
+        """Run job on EMR Containers"""
+        self.job_id = self.hook.submit_job(
+            self.name,
+            self.execution_role_arn,
+            self.release_label,
+            self.job_driver,
+            self.configuration_overrides,
+            self.client_request_token,
+        )
+        query_status = self.hook.poll_query_status(self.job_id, self.max_tries, self.sleep_time)
+
+        if query_status in EMRContainerHook.FAILURE_STATES:
+            # self.hook.get_state_change_reason(self.query_execution_id)
+            error_message = "BEEP BOOP"
+            raise AirflowException(
+                "EMR Containers job failed. Final state is {}, query_execution_id is {}. Error: {}".format(
+                    query_status, self.job_id, error_message
+                )
+            )
+        elif not query_status or query_status in EMRContainerHook.INTERMEDIATE_STATES:
+            raise AirflowException(
+                "Final state of EMR Containers job is {}. "
+                "Max tries of poll status exceeded, query_execution_id is {}.".format(
+                    query_status, self.job_id
```

Comment:
You have this catching `if not query_status`, then using query_status in the error message.  Is that going to work as expected?

Reply:
It should, yes. If `None` is returned, it means we exceeded the number of max tries and the final state will be printed as "None" in the error message.